rs/consensus/src/consensus/finalizer.rs

```diff
@@ -133,6 +134,13 @@ impl Finalizer {
                         .observe(now.duration_since(last_batch_delivered_at).as_secs_f64());
                 }
                 self.last_batch_delivered_at.borrow_mut().replace(now);
+                // Batch creation time is essentially wall time (on some replica), so the median
+                // duration across the subnet is meaningful.
+                self.metrics.finalization_latency.observe(
```

Comment:
What I mean is that the finalization happens in a different iteration of the consensus' event loop. 
Either in the [aggregate](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/consensus/src/consensus.rs?L494) step, if there are enough finalization shares in the replica's consensus pool; or in the [validate](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/consensus/src/consensus.rs?L500) step if the rest of the subnet have already finalized the block, produced the Finalization artifact, and delivered it to the replica.

The Finalizer component is responsible for creating finalization shares & delivering already finalized blocks.

On top of that the batch delivery sometimes fails because the batch queue is full (I mean this error https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/interfaces/src/messaging.rs?L13-14), which contributes to a lag between the finalization and the delivery of a block.

I'd suggest renaming the metric, if you are okay with it

Reply:
I guess we deliver a `Batch` that was derived from the `Block`, so "batch delivery" makes sense to me and is consistent with the method name.